### DISCUSSION TITLE
Setting up the project on android studio 4.1.2

### DISCUSSION THREAD

**Discussion by epicadk** [google/android-fhir#312] [General]
I cloned the repository however when I open the project it throws an error saying 
`Gradle sync failed: This version of the Android Support plugin for IntelliJ IDEA (or Android Studio) cannot open this project, please retry with version 4.2 or newer.` 
I tried invalidating cache and restarting however it still throws the same error. Am I missing a setup while setting up the project or is it an issue with the IDE?  

As a workaround I am using android studio artic fox (4.2 canary) and I also had to update the gradle plugin version to 7.0.0-alpha9 ( because the current gradle plugin version is incompatible with Android studio 4.2)

**Comment by prakharagarwal-dev** [comment]
Same issue occured with me also, so I also built it in 4.2 canary version and updated gradle plugin version to 7.0.0-alpha9. Now it is working fine.

**Comment by s-ayush2903** [comment]
Yes it is so, because of the android gradle plugin that is being used. And to run any project, your AGP should be more than or equal to the one written in the project gradle file. And for this project it is `4.2.0-beta02`, see:
https://github.com/google/android-fhir/blob/b338ed4fdad4bf61fec751fc4a6a4001fb7d8c7f/deps.gradle#L15

**Reply by epicadk** [reply]
> Yes it is so, because of the android gradle plugin that is being used. And to run any project, your AGP should be more than or equal to the one written in the project gradle file. And for this project it is `4.2.0-beta02`, see:
> https://github.com/google/android-fhir/blob/b338ed4fdad4bf61fec751fc4a6a4001fb7d8c7f/deps.gradle#L15

Yes thanks 🤟, but it still doesn't answer my question 😅. Is there a way to build the project in current stable android studio versions or do we have to use the canary/ beta versions?

**Reply by s-ayush2903** [reply]
I'm sorry if the middle line couldn't clarify the answer, putting it more clearly in simpler terms, canary versions should be used to build this project, it doesn't work with otherwise.

**Reply by epicadk** [reply]
It still did help a lot though I was only confused because it doesn't with canary either unless you update the gradle version to `7.0.0-alpha9`. Thankyou 😅

**Comment by jingtang10** [comment]
Thanks for the discussion. I added this: https://github.com/google/android-fhir/wiki/Developer's-Guide#android-studio in the dev guide in the wiki.

**Reply by epicadk** [reply]
Should be usage section under the ReadMe also be updated? 
https://github.com/google/android-fhir#usage
I think that would also cause confusion since it implies that any version above 4.0 would work.

**Comment by deepankarb** [comment]
I am on AS 4.2 beta 3 and due to version incompatibility I am forced to `git stash` when switching branches, can't use `git add -u` (update all added files) - have to be careful about commits.

Going forward, I think we should stick to the latest stable AGP and gradle versions to cause [least confusion](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) to developers.

**Reply by epicadk** [reply]
> I am on AS 4.2 beta 3 and due to version incompatibility I am forced to `git stash` when switching branches, can't use `git add -u` (update all added files) - have to be careful about commits.
> 
> Going forward, I think we should stick to the latest stable AGP and gradle versions to cause [least confusion](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) to developers.

Yup thanks. I thought it was just me.

**Reply by epicadk** [reply]
@deepankarb I think we can switch to  AGP 4.2.0 now, it would be compatible with the latest stable release of android studio as well.

**Comment by Abhinavtdk** [comment]
When I tried to run it, I got the following error:
''Please remove usages of `jcenter()` Maven repository from your build scripts and migrate your build to other Maven repositories.''

**Reply by epicadk** [reply]
> When I tried to run it, I got the following error:
> ''Please remove usages of `jcenter()` Maven repository from your build scripts and migrate your build to other Maven repositories.''

 I think that's a warning and not an error.

**Reply by epicadk** [reply]
If you are having an issue setting up can you share your stacktrace?
